### PR TITLE
Create a desktop shortcut

### PR DESCRIPTION
**Pull request by jkeiser**
1. Includes the outstanding PR to fix up the existing start menu shortcut
2. Adds a desktop shortcut
3. Uses appropriate icons for both instead of scaling up the 16x16 version.

### CODE REVIEW

File: config/projects/chefdk.rb [hunk 1]
```diff
@@ -99,7 +99,7 @@
 
 package :msi do
   upgrade_code "AB1D6FBD-F9DC-4395-BDAD-26C4541168E7"
-  signing_identity "F74E1A68005E8A9C465C3D2FF7B41F3988F0EA09", machine_store: true
+#  signing_identity "F74E1A68005E8A9C465C3D2FF7B41F3988F0EA09", machine_store: true
```
Comment (chefsalim) [inline config/projects/chefdk.rb]: Needs to be reverted so the package is properly signed

File: resources/chefdk/msi/localization-en-us.wxl.erb [hunk 2]
```diff
@@ -20,9 +20,12 @@
...
-  
-  <String Id="ChefDkStartMenuShortcutDefName"><%= friendly_name %></String>
-  <String Id="ChefDkStartMenuShortcutDefDescription"><%= friendly_name %> Command Prompt</String>
+
+  <String Id="ChefDkShortcutDefName"><%= friendly_name %></String>
+  <String Id="ChefDkShortcutDefDescription"><%= friendly_name %> Command Prompt</String>
+
   <String Id="FeatureChefDkStartMenuShortcut">Start Menu Shortcut</String>
-  <String Id="FeatureChefDkStartMenuShortcutDescription">Install a shortcut on the start menu</String>
+  <String Id="FeatureChefDkStartMenuShortcutDescription">Install a ChefDK shortcut on the user's start menu. The shortcut will launch powershell (or conemu, if installed) with the ChefDK in the PATH and ChefDK powershell extensions in it.</String>
```
Comment (chefsalim) [inline resources/chefdk/msi/localization-en-us.wxl.erb]: change capitalization: "PowerShell" and "ConEmu"

I actually like the original simpler wording.. "Install a shortcut on the Start Menu".

File: resources/chefdk/msi/localization-en-us.wxl.erb [hunk 3]
```diff
@@ -20,9 +20,12 @@
...
+  <String Id="FeatureChefDkDesktopShortcut">Desktop Shortcut</String>
```
Comment (chefsalim) [inline resources/chefdk/msi/localization-en-us.wxl.erb]: Last minute feature creep :) .. would be better to leave desktop shortcut for a separate PR at this point.

File: resources/chefdk/msi/source.wxs.erb [hunk 4]
```diff
@@ -65,13 +65,26 @@
...
         </Directory>
       </Directory>
-      
+
       <Directory Id="ProgramMenuFolder">
         <Component Id="StartMenuShortcut" Guid="{6DD3FFF3-E009-40E7-B1C2-EF606B941CCD}">
-            <Shortcut Id="StartMenuShortcutDef" 
-                      Name="!(loc.ChefDkStartMenuShortcutDefName)"
-                      Description="!(loc.ChefDkStartMenuShortcutDefDescription)"
-                      Target="[#StartChefDkScript]"/>
```
Comment (chefsalim) [inline resources/chefdk/msi/source.wxs.erb]: Need to confirm what the # is actually doing.. I see examples without the # in the WIX books.

File: resources/chefdk/msi/localization-en-us.wxl.erb (hunk 2 above)
Comment (jkeiser) [inline resources/chefdk/msi/localization-en-us.wxl.erb]: I'm OK reverting to the simpler wording, but we do get a whole screen to put our narrative on, so it probably doesn't hurt to explain what it is you're going to get to people who want to know IMO.

File: resources/chefdk/msi/localization-en-us.wxl.erb (hunk 3 above)
Comment (jkeiser) [inline resources/chefdk/msi/localization-en-us.wxl.erb]: There is a separate PR if you want to look at that one instead :)

Comment (jkeiser) [inline resources/chefdk/msi/localization-en-us.wxl.erb]: https://github.com/chef/omnibus-chef/pull/479

File: resources/chefdk/msi/source.wxs.erb (hunk 4 above)
Comment (jkeiser) [inline resources/chefdk/msi/source.wxs.erb]: Do you consider this a blocker? I'd like to not go through a build cycle again if I can avoid it.

Comment (chefsalim) [inline resources/chefdk/msi/source.wxs.erb]: Actually, "#" seems like the correct usage for this case - never mind.

File: resources/chefdk/msi/localization-en-us.wxl.erb [hunk 5]
```diff
@@ -20,9 +20,12 @@
...
-  
-  <String Id="ChefDkStartMenuShortcutDefName"><%= friendly_name %></String>
-  <String Id="ChefDkStartMenuShortcutDefDescription"><%= friendly_name %> Command Prompt</String>
+
+  <String Id="ChefDkShortcutDefName"><%= friendly_name %></String>
+  <String Id="ChefDkShortcutDefDescription"><%= friendly_name %> Command Prompt</String>
+
   <String Id="FeatureChefDkStartMenuShortcut">Start Menu Shortcut</String>
-  <String Id="FeatureChefDkStartMenuShortcutDescription">Install a shortcut on the start menu</String>
+  <String Id="FeatureChefDkStartMenuShortcutDescription">Install a ChefDK shortcut on the user's start menu.</String>
```
Comment (chefsalim) [inline resources/chefdk/msi/localization-en-us.wxl.erb]: change to "Start Menu"

**Comment by chefsalim** [comment]
Looks good! :1

File: resources/chefdk/msi/assets/start-chefdk.ps1 [hunk 6]
```diff
@@ -1,6 +1,8 @@
 Try {
     $conemulocation = "$env:programfiles\ConEmu\Conemu64.exe"
-    $chefdkinit = 'chef shell-init powershell | iex; Import-Module chef; $env:CHEFDK_ENV_FIX = 1'
+    # We don't want the current path to affect which "chef shell-init powershell" we run, so we need to set the PATH to include the current omnibus.
+    $chefdk_bin = (split-path $MyInvocation.MyCommand.Definition -Parent)
+    $chefdkinit = '"$env:PATH = ''' + $chefdk_bin + ';'' + $env:PATH; $env:CHEFDK_ENV_FIX = 1; chef shell-init powershell | out-string | iex; $OldWarningPreference = $WarningPreference; $WarningPreference = ''SilentlyContinue''; Import-Module chef; $WarningPreference = $OldWarningPreference"'
```
Comment (mwrock) [inline resources/chefdk/msi/assets/start-chefdk.ps1]: Are you suppressing warnings to avoid warnings about the cmdlet verbs? You could call Import-Module with -DisableNameChecking to do that.

**Comment by mwrock** [comment]
LGTM :+1:
